server/game/cards/events/02/loot.js

```diff
@@ -0,0 +1,73 @@
+const _ = require('underscore');
+
+const DrawCard = require('../../../drawcard.js');
+
+class Loot extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                afterChallenge: (event, challenge) => (
+                    this.controller === challenge.winner
+                    && challenge.isUnopposed()
+                    && this.opponentHasGold()
+                    && this.opponentHasDrawDeck())
+            },
+            handler: () => {
+                let opponent = this.game.getOtherPlayer(this.controller);
+
+                if(!opponent) {
+                    return false;
+                }
+
+                let maxDiscard = Math.min(opponent.gold, opponent.drawDeck.size());
+                let buttons = _.map(_.range(1, maxDiscard + 1), num => {
+                    return { text: num, method: 'numberSelected', arg: num };
+                });
+
+                this.game.promptWithMenu(this.controller, this, {
+                    activePrompt: {
+                        menuTitle: 'Select # of cards to discard',
+                        buttons: buttons
+                    },
+                    source: this
+                }); 
+            }
+        });
+    }
+
+    numberSelected(player, num) {
+        let opponent = this.game.getOtherPlayer(this.controller);
+
+        opponent.discardFromDraw(num);
+        opponent.gold -= num;
```

Comment:
This should be this.game.addGold(opponent, -num) so that events trigger properly for them spending gold